### DISCUSSION TITLE
How to specify a constraint for a bunch of CheckBox that at least a CheckBox should be checked?

### DISCUSSION THREAD

**Discussion by evan0greenup** [slint-ui/slint#2932] [Q&A]
When there is a bunch of CheckBox (Just mentally selected, no extra components are involved), If I want to add a logical constraint to them which at any moment, at least a CheckBox should be checked.

What is the most efficient way to achieve that?

**Comment by ogoffart** [comment]
One day we'll have a RadioButton (https://github.com/slint-ui/slint/issues/2732)

In the mean time, it has to be done manually.
For example like this: 

https://github.com/slint-ui/slint/blob/c170c517a3b0934260fc53de78003925676e0e9f/examples/fancy_demo/main.slint#L388-L395

In that example, the RadioButton is just like a CheckBox, but it uses "clicked" instead of "toggled"

And also the unsel funciton is https://github.com/slint-ui/slint/blob/c170c517a3b0934260fc53de78003925676e0e9f/examples/fancy_demo/main.slint#L363
